maryna-karpovets/nodejs-aws-be
# Task 9. Add backend for frontend with elastic beanstalk

Task 9 [https://github.com/rolling-scopes-school/nodejs-aws-tasks/blob/main/task9-bff/task.md](https://github.com/rolling-scopes-school/nodejs-aws-tasks/blob/main/task9-bff/task.md)

All tasks and additional ones are done.

Elastik beanstalk **http://maryna-karpovets-bff-api-bff-dev-env.eu-west-1.elasticbeanstalk.com/**

To test run in postman requests for products or cart like (Do not use the body for `GET` requests as it was said in the lecture, it should be none):
- GET  `http://maryna-karpovets-bff-api-bff-dev-env.eu-west-1.elasticbeanstalk.com/products`
- POST `http://maryna-karpovets-bff-api-bff-dev-env.eu-west-1.elasticbeanstalk.com/products`
**body**:
```
{
    "title": "Christmas trip",
    "description": "Best Christmas",
    "price": 1000,
    "count": 11
}
```
- GET `http://maryna-karpovets-bff-api-bff-dev-env.eu-west-1.elasticbeanstalk.com/cart`

3 - A working and correct nest application, works as described in task 9.1.
5 - The bff-service is deployed with Elastic Beanstalk. The bff-service call is redirected to the appropriate service: product-service or CART.

Additional (optional) tasks

+1 - Add a cache at the bff-service level for a request to the getProductsList function of the product-service. The cache should expire in 2 minutes.
How to test:
Get products list `http://maryna-karpovets-bff-api-bff-dev-env.eu-west-1.elasticbeanstalk.com/products`
Create a new product (the description with the body above)
Get products list - the result shouldn’t have a new product
Wait more than 2 minutes
Get products list - the result should have a new product
+1 - Use NestJS to create bff-service instead of express


Score 7 / 7
